Assets/Scripts/UI/FadePopup.cs

```diff
@@ -0,0 +1,38 @@
+using Kuroneko.UIDelivery;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+
+namespace UI
+{
+    public class FadePopup : Popup
+    {
+        [SerializeField] private Image fadeImage;
+        [SerializeField] private TextMeshProUGUI dayText;
+        
+
+        protected override void InitPopup()
+        {
+            DayNight.OnDayChange += OnDayChange;
+        }
+
+        private void OnDayChange(float currentAlpha, float targetAlpha)
+        {
+            dayText.text = "DAY " + DayNight.currentDay;
+
+            LeanTween.alpha(fadeImage.rectTransform, targetAlpha, DayNight.duration); // gradually turn Alpha of Image
+            LeanTween.value(gameObject, currentAlpha, targetAlpha, DayNight.duration)
+            .setOnUpdate((float alpha) => {
+                Color color = dayText.color;
+                color.a = alpha;
+                dayText.color = color;
+            }); // gradually turn Alpha of Text
+        }
```

Comment:
Ideally, I want this OnDayChange to handle the fade in then fade out in one function.
1. Fade in the black panel
2. Show the day text change
3. Fade out the black panel

To make things easier, you can put the day text as a child of the fade image. So when the fade image's alpha changes, the day text's alpha also changes.

```suggestion
        const FADE_DURATION = 2.0f;
        const WAIT_DURATION = 1.0f;

        private void OnDayChange(float previousDay, float nextDay)
        {
            dayText.text = string.Empty;
            // Fade In
            LeanTween.alpha(fadeImage.rectTransform, 0f, 1f, FADE_DURATION).setOnComplete(() => {
                   dayText.text = "DAY " + previousDay;
            });
            
            // Wait a bit to transition the text
            LeanTween.wait(WAIT_DURATION).setOnComplete(() => {
                  dayText.text = "DAY " + nextDay;
            });
            
            // Fade Out
            LeanTween.alpha(fadeImage.rectTransform, 1f, 0f, FADE_DURATION).setOnComplete(() => {
                DayNight.StartNewDay();
            });
        }
```

Reply:
Hi, I putted "day text" in "FadePanel" and used above code, but it didn't change the alpha for both of them. Please tell me if I'm doing something wrong. 

<img width="250" alt="Screenshot 2025-03-04 at 13 48 03" src="https://github.com/user-attachments/assets/62747f7d-5ff2-4aaf-923c
<img width="648" alt="Screenshot 2025-03-04 at 13 48 19" src="https://github.com/user-attachments/assets/92a69826-25fa-4b74-98dc-79c4f49bbd15" />
-fd44bad5f414" />

Reply:
Hi, I made "day text" as a child of "fadePanel", but above code didn't change both of alpha. Please tell me if I'm doing something wrong. 
<img width="668" alt="Screenshot 2025-03-04 at 13 53 15" src="https://github.com/user-attachments/assets/56d2e3fc-6a4f-4816-b980-c165
<img width="253" alt="Screenshot 2025-03-04 at 13 53 36" src="https://github.com/user-attachments/assets/4570e035-c496-4372-b64f-a2585d2c852a" />
64ea59dc" />

Reply:
Looks like it will work. I hate how we are using nested callbacks. I couldn't find any way to sequence tweens on LeanTween. Happy to approve this when you make the code changes.

Reply:
Alright, then I'll use this code and commit it once, but I'll also try to find if there is any other efficient way to solve this problem, while I do the weather system. Thanks